src/ServicePulse.Host/app/js/views/monitored_endpoints/monitored_endpoints.controller.js

```diff
@@ -47,6 +54,44 @@
                     // Warn user
                 });
             });
+
+            ednpointsFromScSubscription = createSourceFrom(function () {
+                return serviceControlService.getExceptionGroupsPromise('Endpoint Name', null);
+            }).subscribe(function (endpoint) {
+                var index = $scope.endpoints.findIndex(function (item) { return item.name === endpoint.title });
+                if (index >= 0) {
+                    $scope.endpoints[index].errorCount = endpoint.count;
+                } else {
+                    $scope.endpoints.push({ name: endpoint.title, errorCount: endpoint.count, isConnected: false });
+                }
+            });
+        }
+
+        function createSourceFrom(promise) {
```

Comment:
THat is a stab at not mixing RX into service code. I could extract it as a separate service in which case any RX interaction would be made 
`rxify.CreateSource(function(){}).subscribe..`
Whever function need to return promise. @WilliamBZA let me know what do you think of such approach. It is nothing more than experiment.